### ISSUE TITLE
Pasting table cells from clipboard produces malformed tables

### ISSUE THREAD

**Issue by bradleyayers** [ProseMirror/prosemirror-tables#41]
When copy/pasting a table, the first cell (top left) is empty, and its content is inserted before the table. This regression was introduced in 26c640f7bfcdeafe639440fcc2f25f1035c52a09.

Demo:

![2018-01-23 prosemirror-tables-41](https://user-images.githubusercontent.com/105820/35243251-499d3490-0010-11e8-9e7b-fea58c252164.gif)

(What is perhaps the same problem) also manifests as an exception when pasting a single cell. The exception started occurring in 8fee4bb.

Demo:

![2018-01-23 prosemirror-tables-41 single cell](https://user-images.githubusercontent.com/105820/35244005-d986b138-0012-11e8-944b-87a2adfd7a56.gif)

Stack trace:

```
demo_bundle.js:4744 Uncaught TypeError: Cannot read property 'content' of null
    at fitRightClosed (demo_bundle.js:4744)
    at fitRightClosed (demo_bundle.js:4752)
    at fitRightJoin (demo_bundle.js:4733)
    at fitRightJoin (demo_bundle.js:4711)
    at fitRight (demo_bundle.js:4778)
    at replaceStep (demo_bundle.js:4621)
    at Transaction.Transform.replace (demo_bundle.js:4640)
    at Transaction.Transform.replaceRange (demo_bundle.js:5030)
    at TextSelection.replace (demo_bundle.js:5219)
    at TextSelection.replace (demo_bundle.js:5399)
fitRightClosed @ demo_bundle.js:4744
fitRightClosed @ demo_bundle.js:4752
fitRightJoin @ demo_bundle.js:4733
fitRightJoin @ demo_bundle.js:4711
fitRight @ demo_bundle.js:4778
replaceStep @ demo_bundle.js:4621
Transform.replace @ demo_bundle.js:4640
Transform.replaceRange @ demo_bundle.js:5030
replace @ demo_bundle.js:5219
replace @ demo_bundle.js:5399
replaceSelection @ demo_bundle.js:5746
doPaste @ demo_bundle.js:9542
editHandlers.paste @ demo_bundle.js:9549
view.dom.addEventListener.view.eventHandlers.(anonymous function) @ demo_bundle.js:9129
```

/cc @marijnh

**Comment by marijnh** [comment]
Are you testing with prosemirror-model 1.1.0 and prosemirror-view 1.07 (or higher)? Those should, in theory, address at least the first problem.

**Comment by bradleyayers** [comment]
I believe I'm running with the latest

- prosemirror-model@1.1.0
- prosemirror-view@1.0.9

I'm testing using this repo and the command:

```
git pull && \
rm -rf node_modules/ && \
npm install && \
npm run build_demo && \
open index.html
```

**Comment by marijnh** [comment]
Oh, right, the issue I was thinking of was similar but for content pasted from outside of the editor itself. If you get prosemirror-transform 1.0.2, the crash should be fixed, and with prosemirror-view 1.0.10, cell selections should remain properly closed when pasted. Let me know if that helps.

**Comment by bradleyayers** [comment]
Excellent! Thank-you very much!

I've tested with those two new versions and all is well!
